custom_components/ppc_smgw/gateways/theben/thebenconnexa/thebenconnexa.py

```diff
@@ -0,0 +1,73 @@
+import httpx
+
+from custom_components.ppc_smgw.gateways.reading import Information, OBISCode, Reading
+from datetime import datetime
+
+
+class ThebenConnexaClient:
+    def __init__(
+        self,
+        base_url: str,
+        username: str,
+        password: str,
+        httpx_client: httpx.AsyncClient,
+        logger,
+    ):
+        self.base_url = base_url
+        self.username = username
+        self.password = password
+
+        self.httpx_client = httpx_client
+        self.logger = logger
+
+        self.httpx_client.headers.setdefault("Content-Type", "application/json")
```

Comment:
```suggestion
        self.httpx_client.headers.setdefault("Content-Type", "application/json")
        self.httpx_client.follow_redirects = True
```

Following redirects here might be a good idea because 
`https://192.168.xx.xx/smgw/m2m/` redirects to `https://192.168.33.2/smgw/m2m/<userid>.sm/json`.
Without redirect users need to set the full URL with <userid> already in the base_url.

Reply:
Yes, it does work locally and it also works with the integration when including the `json` fix.

Reply:
Let's try with the fixes :)

According to section A-2.1 of the documentation, users don't need to specify the userid themselves. Is this the case for you?

Reply:
Yes no need to specify the userid when`follow_redirects = True`. The httpx client will do 3 requests automatically:
- the first will return 401 with the information for the correct digest auth connfiguration in the WWW-Authenticate header (realm, qop, nonce, etc.)
- the second then is returning 307 due to the redirect
- the third is successfully querying the information